uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/view/items/AbstractDockItem.java

```diff
@@ -132,9 +138,29 @@ public String getLabel() {
 
     public abstract void close();
 
-    void setupCSSLocators(UberfireDock dock) {
-        getElement().addClassName(CSSLocatorsUtils.buildLocator("qe-docks-item",
+    void setupLocators(final UberfireDock dock) {
+        initOuiaComponentAttributes();
+        getElement().addClassName(CSSLocatorsUtils.buildLocator(OUIA_COMPONENT_TYPE,
                                                                 dock.getDockPosition().getShortName(),
                                                                 dock.getIdentifier()));
     }
+
+    @Override
+    public Consumer<OuiaAttribute> ouiaAttributeRenderer() {
+        return ouiaAttribute -> getElement().setAttribute(ouiaAttribute.getName(), ouiaAttribute.getValue());
+    }
+
+    @Override
+    public OuiaComponentTypeAttribute ouiaComponentType() {
+        return new OuiaComponentTypeAttribute(OUIA_COMPONENT_TYPE);
+    }
+
+    @Override
+    public OuiaComponentIdAttribute ouiaComponentId() {
+        return new OuiaComponentIdAttribute(OUIA_COMPONENT_TYPE +
+                                                    "-" +
+                                                    dock.getDockPosition().getShortName() +
```

Comment:
imo we don't need for the id to keep all the hierarchy (I assume that the first variable is specifying the parent element, correct me if I am wrong) - `OUIA_COMPONENT_TYPE + "-" + "dock.getIdentifier()"` should be enough, you'll typically navigate in tests by:
`cy.get(dock-by-ouia-id).get(item-by-ouia-id)` so the item-by-ouia-id selector needs to be unique just in context of given dock (thus we can save some letters in name). 

Though, based on your preference and rules you want to set for the project.

Reply:
agree, will remove this part